### ISSUE TITLE
ODE bugs

### ISSUE THREAD

**Issue by kquine** [dreal/dreal3#112]
I am using the version `dReal-3.15.05.01-darwin`. 
The following gives `unsat`, but should be `sat`.

```
(set-logic QF_NRA_ODE)
(declare-fun x1 () Real)
(declare-fun x2 () Real)
(define-ode flow_1 ((= d/dt[x1] 0.0)(= d/dt[x2] 1.0)))
(declare-fun x1_0 () Real)
(declare-fun x1_t () Real)
(declare-fun x2_0 () Real)
(declare-fun x2_t () Real)
(declare-fun time_0 () Real)
(assert (= time_0 1))
(assert (= x1_0 0))
(assert (= x1_t 0))
(assert (= x2_0 0))
(assert (= x2_t 1))
(assert (= [x1_t x2_t] (integral 0. time_0 [x1_0 x2_0] flow_1)))
(check-sat)
(exit)
```

If we slightly change the bound of `time_0`, then it returns `sat`.

```
(set-logic QF_NRA_ODE)
(declare-fun x1 () Real)
(declare-fun x2 () Real)
(define-ode flow_1 ((= d/dt[x1] 0.0)(= d/dt[x2] 1.0)))
(declare-fun x1_0 () Real)
(declare-fun x1_t () Real)
(declare-fun x2_0 () Real)
(declare-fun x2_t () Real)
(declare-fun time_0 () Real)
(assert (and (<= 0.9999 time_0) (<= time_0 1)))
(assert (= x1_0 0))
(assert (= x1_t 0))
(assert (= x2_0 0))
(assert (= x2_t 1))
(assert (= [x1_t x2_t] (integral 0. time_0 [x1_0 x2_0] flow_1)))
(check-sat)
(exit)
```

In addition, the following gives a CAPD exception:

```
(set-logic QF_NRA_ODE)
(declare-fun x1 () Real)
(declare-fun x2 () Real)
(define-ode flow_1 ((= d/dt[x1] 0.0)(= d/dt[x2] 1.0)))
(declare-fun x1_0 () Real)
(declare-fun x1_t () Real)
(declare-fun x2_0 () Real)
(declare-fun x2_t () Real)
(declare-fun time_0 () Real)
(assert (= time_0 1))
(assert (and (<= x1_0 0) (<= x1_0 1)))
(assert (and (<= x1_t 0) (<= x1_t 1)))
(assert (and (<= x2_0 0) (<= x1_0 1)))
(assert (and (<= x2_t 1) (<= x2_t 2)))
(assert (= [x1_t x2_t] (integral 0. time_0 [x1_0 x2_0] flow_1)))
(check-sat)
(exit)
```

This gives the following exception:

```
libc++abi.dylib: terminating with uncaught exception of type std::logic_error: C0DoubletonSet fatal error! Intersection of two enclosures of the same object is empty. Report this error to CAPD developers!
```

**Comment by soonhokong** [comment]
@kquine, thanks! I'll investigate it.

**Comment by soonhokong** [comment]
Thanks again for the report. The first problem is fixed by 0583f16. The third example does not have proper bounds and pass an interval with infinity and causes a problem. I changed it to the following and it fixed the problem.

```lisp
...
(assert (and (<= 0 x1_0) (<= x1_0 1)))
(assert (and (<= 0 x1_t) (<= x1_t 1)))
(assert (and (<= 0 x2_0) (<= x2_0 1)))
(assert (and (<= 1 x2_t) (<= x2_t 2)))
...
```

**Comment by soonhokong** [comment]
> The third example does not have proper bounds and pass an interval with infinity and causes a problem.

Of course, it is still a problem because CAPD generates the exception for the case. I'll ask CAPD authors about this. At least, we can catch the exception from CAPD and do branching and proceed.
